src/mower_logic/src/mower_logic/behaviors/MowingBehavior.cpp

```diff
@@ -363,6 +364,13 @@ bool MowingBehavior::execute_mowing_plan() {
                         skip_area = false;
                         return true;
                     }
+                    if(skip_path) 
+                    {
+                        currentMowingPaths.erase(currentMowingPaths.begin());
+                        skip_path=false;
+                        return true;
```

Comment:
should return false

Reply:
Are you sure? It also returns true when doing a skip area?

Reply:
yes, look at the end of this method - true means all paths complete. Can either return false here or return `currentMowingPaths.empty();`

Reply:
check, fixed it